test/e2e/framework/nodes.go

```diff
@@ -59,6 +65,31 @@ func findNodesByGatewayLabel(cluster int, isGateway bool) []*v1.Node {
 	return retNodes
 }
 
+// FindMultipleGatewaysCluster finds the cluster with multiple GW nodes.
+// Returns cluster index and active GW Pod and Node names.
+func (f *Framework) FindMultipleGatewaysCluster(clusters []string) (int, map[string]string) {
+	activeGW := make(map[string]string)
+
+	for idx := range clusters {
+		gatewayNodes := findNodesByGatewayLabel(idx, true)
+		if len(gatewayNodes) >= 2 {
+			cluster := idx
+			gwPods := f.AwaitPodsByAppLabel(ClusterIndex(cluster), SubmarinerGateway, TestContext.SubmarinerNamespace, -1)
+
+			for i := range gwPods.Items {
+				if gwPods.Items[i].Labels[gatewayStatusLabel] == gatewayStatusActive {
+					activeGW["Pod"] = gwPods.Items[i].Name
+					activeGW["Node"] = gwPods.Items[i].Labels[gatewayNodeLabel]
+
+					return cluster, activeGW
+				}
+			}
```

Comment:
It's cleaner to return the pod and node directly rather than embedded in a map with well-known keys. However, perhaps we shouldn't overload this function to return the active GW as well. Instead I suggest we add an `AwaitActiveGatewayPod` function (in _pods.go_):

```Go
func (f *Framework) AwaitActiveGatewayPod(cluster ClusterIndex, checkPod func(*v1.Pod) bool) *v1.Pod {
    for retries := 1; retries <= 30; retries++ {
        var activePod, retPod *v1.Pod

        gwPods := f.AwaitPodsByAppLabel(ClusterIndex(cluster), SubmarinerGateway, TestContext.SubmarinerNamespace, -1)
	for i := range gwPods.Items {
	    pod := &gwPods.Items[i]
	    if pod.Labels[gatewayStatusLabel] == gatewayStatusActive {
	        if activePod != nil {
		    Errorf("Found 2 active gateway pods: %q and %q", activePod.Name, pod.Name)
		    retPod = nil
		    break
		}

		activePod = pod

		if checkPod == nil || checkPod(pod) {
		    retPod = pod
		}
	    }
        }

        if retPod != nil {
            return retPod
        }

        time.Sleep(2 * time.Second)
    }

    return nil
}
```

Then https://github.com/submariner-io/submariner/pull/2014 can call this to get the initial active GW:

```Go
primaryCluster := f.FindMultipleGatewaysCluster(framework.TestContext.ClusterIDs)
...
initialGWPod := f.AwaitActiveGatewayPod(primaryCluster, nil)
Expect(initialGWPod).ToNot(BeNil(), "Did not find an active gateway pod")
```

And to get the new GW pod after failover:

```Go
newGWPod := f.AwaitActiveGatewayPod(primaryCluster, func(pod *v1.Pod) bool {
    return pod.Spec.NodeName != initialGWPod.Spec.NodeName
})
```

Here we just return in cluster index.

Reply:
@tpantelis 

In the current function I'm also returning the GW node name.
Is it ok to return the gw node name in `AwaitActiveGatewayPod` function or better to create a dedicated function that will return the node name of gw node?

Reply:
The node name can be extracted from the `Pod`, ie `pod.Spec.NodeName`.

Reply:
Got it.
Thanks.

Reply:
Thank you for the idea and example.
That's a great code re use.